### ISSUE TITLE
Missing default exports cause warnings instead of errors when useTypescriptIncrementalApi is true 

### ISSUE THREAD

**Issue by nfm** [TypeStrong/fork-ts-checker-webpack-plugin#315]
## Current behavior

When using the [fork-ts-checker-webpack-plugin example](https://github.com/TypeStrong/ts-loader/tree/master/examples/fork-ts-checker-webpack-plugin), attempting to import a missing default export only causes a warning.

If `useTypescriptIncrementalApi` is set to `false`, the build errors as expected.

## Expected behavior

The build should error when a default export is missing.

## Steps to reproduce the issue

1. Copy the [fork-ts-checker-webpack-plugin example](https://github.com/TypeStrong/ts-loader/tree/master/examples/fork-ts-checker-webpack-plugin)
2. Edit `src/index.tsx` to introduce an import of a missing default export:
```diff
import * as React from 'react';
import * as ReactDOM from 'react-dom';
- import { App } from './app';
+ import App from './app';
```
3. Run `yarn run build`. The build will succeed (exit code 0) with the following warning printed:
```
WARNING in ./src/index.tsx 4:36-39
"export 'default' (imported as 'App') was not found in './app'
 @ multi core-js whatwg-fetch ./src/index.tsx
```
4. Edit `webpack.config.production.js` to set `useTypescriptIncrementalApi: false`
5. Run `yarn run build`. The build errors (exit code 2) with the following error printed:
```
ERROR in /tmp/ts-loader/examples/fork-ts-checker-webpack-plugin/src/index.tsx
ERROR in /tmp/ts-loader/examples/fork-ts-checker-webpack-plugin/src/index.tsx(3,8):
TS1192: Module '"/tmp/ts-loader/examples/fork-ts-checker-webpack-plugin/src/app"' has no default export.
```

## Issue reproduction repository

Please follow above steps.

## Environment

Reproduces using the deps from the example repo above.

In our codebase, we hit this issue on:

 - **fork-ts-checker-webpack-plugin**: 1.4.1
 - **typescript**: 3.4.5
 - **tslint**: 5.16.0
 - **webpack**: 4.32.2
 - **os**: Ubuntu 19.04, also reproduces for other team members on Mac OS

---

This issue possibly relates to the re-export issues documented on [ts-loader](https://github.com/TypeStrong/ts-loader/blob/218718aac3cc1ebf668181e2e87047d276a1cf15/README.md#transpileonly-boolean-defaultfalse).

I'm not familiar enough with webpack plugins or the typescript compiler to dig into this much further, but am happy to assist if there's more info I can provide. Thanks a lot! :heart:

**Comment by johnnyreilly** [comment]
That is indeed strange!

Question: does a proper build error fail the build with `useTypescriptIncrementalApi: true`?

eg `const name: string = 1;`

I'm assuming yes but wanted to double check. I'm curious as to why this is classified as a warning but not an error. Warnings should still fail the build to my mind

Looking at the output, it seems that it's webpack reporting the issue (which given it's the bundler that makes sense). I'm not sure why a warning from webpack isn't failing the build.

Could you do some digging into how to ensure webpack builds fail when a warning presents please? They should fail in the case of warnings and so I don't know why they would not be.

**Comment by nfm** [comment]
Thanks @johnnyreilly.

> Question: does a proper build error fail the build with useTypescriptIncrementalApi: true?
> eg const name: string = 1;

Yes, this does fail in the example repo (for both `useTypescriptIncrementalApi: true` and `useTypescriptIncrementalApi: false`).

```patch
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { App } from './app';
+ 
+ const name: string = 1;

ReactDOM.render(<App />, document.getElementById('root'));
```

```bash
$ yarn run build
<snip>
ERROR in /home/nfm/repos/projects/ts-loader/examples/fork-ts-checker-webpack-plugin/src/index.tsx
ERROR in /home/nfm/repos/projects/ts-loader/examples/fork-ts-checker-webpack-plugin/src/index.tsx(5,7):
TS2322: Type '1' is not assignable to type 'string'.

ERROR in /home/nfm/repos/projects/ts-loader/examples/fork-ts-checker-webpack-plugin/src/index.tsx
ERROR in /home/nfm/repos/projects/ts-loader/examples/fork-ts-checker-webpack-plugin/src/index.tsx(5,7):
TS6133: 'name' is declared but its value is never read.
Child HtmlWebpackCompiler:
     1 asset
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./src/index.html 764 bytes {0} [built]
    [2] (webpack)/buildin/global.js 472 bytes {0} [built]
    [3] (webpack)/buildin/module.js 497 bytes {0} [built]
        + 1 hidden module
error Command failed with exit code 2.
```

I get the same output for both values of `useTypescriptIncrementalApi` there.

Other kinds of import related issues fail the build as expected with `useTypescriptIncrementalApi: true`. For example:

```patch
- import { App } from './app';
+ import { NotTheApp } from './app';
 
- ReactDOM.render(<App />, document.getElementById('root'));
+ ReactDOM.render(<NotTheApp />, document.getElementById('root'));
```

```bash
$ yarn run build
<snip>
ERROR in /home/nfm/repos/projects/ts-loader/examples/fork-ts-checker-webpack-plugin/src/index.tsx
ERROR in /home/nfm/repos/projects/ts-loader/examples/fork-ts-checker-webpack-plugin/src/index.tsx(3,10):
TS2305: Module '"./app"' has no exported member 'NotTheApp'.
Child HtmlWebpackCompiler:
     1 asset
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./src/index.html 764 bytes {0} [built]
    [2] (webpack)/buildin/global.js 472 bytes {0} [built]
    [3] (webpack)/buildin/module.js 497 bytes {0} [built]
        + 1 hidden module
error Command failed with exit code 2.
```

It seems to only be missing default exports that warn instead of error.

It's entirely possible our webpack config has something odd going on, which is why I reached for this example repo for a reproduction :stuck_out_tongue_closed_eyes:. We do unfortunately have some webpack warnings we wouldn't want to fail the build on (including [re-export warnings from ts-loader](https://github.com/TypeStrong/ts-loader/blob/218718aac3cc1ebf668181e2e87047d276a1cf15/README.md#transpileonly-boolean-defaultfalse)), so ideally this issue would present as an error instead of a warning.

I'm not really across how webpack handles plugin warnings vs errors, I might be able to tap a co-worker for a bit more insight there.

**Comment by piotr-oles** [comment]
@nfm
Please try [fork-ts-checker-webpack-plugin@alpha](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/tree/alpha) - I've published a new version which should resolve this issue 🚀
I will close this issue to clean-up the backlog. If this release didn't solve the issue, we can re-open this :)
